libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable, Provider } from '@angular/core';
+import { GoogleAnalyticsService } from 'ngx-google-analytics';
+import { getActionTypeFromInstance, NgxsNextPluginFn, NgxsPlugin, NGXS_PLUGINS } from '@ngxs/store';
+
+/** Primitive types */
+const PRIMITIVE_TYPES = ['bigint', 'boolean', 'number', 'string'];
+
+/**
+ * Determines whether primitive is
+ * @param value
+ * @returns primitive
+ */
+function isPrimitive(value: unknown): value is bigint | boolean | number | string | null | undefined {
+  return value == null || PRIMITIVE_TYPES.includes(typeof value);
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StateAnalyticsPluginService implements NgxsPlugin {
+  /**
+   * Creates an instance of logger plugin.
+   * @param ga
+   */
+  constructor(private readonly ga: GoogleAnalyticsService) {}
+
+  /**
+   * Handles logger plugin
+   * @param state
+   * @param action
+   * @param next
+   * @returns
+   */
+  handle(state: unknown, action: unknown, next: NgxsNextPluginFn) {
+    console.log('In handle');
+    this.logAction(action);
+    return next(state, action);
+  }
+
+  /**
+   * Logs action trigerred to google analytics event
+   * @param action
+   * @returns action
+   */
+  private logAction(action: unknown): void {
+    const type = getActionTypeFromInstance(action);
+    console.log('type', type);
+
+    if (type) {
```

Comment:
We should only log approved actions. Not all of them

Reply:
Not resolved. Type needs to be checked against approved action list/set to determine if the action should be logged